src/background.jl

```diff
@@ -0,0 +1,64 @@
+"""
+    gen_img = generatecanvas(colortype, x, y)
+    gen_img = generatecanvas(colortype, x, y, background)
+    gen_img = generatecanvas(colortype, [x,y])
+    gen_img = generatecanvas(colortype, [x,y], background)
+
+    Creates an image of with the given colortype and size.
+    Defaults to a black solid image
+"""
+generatecanvas(colortype::Type, x::Int, y::Int) = zeros(colortype, (x,y))
+
+generatecanvas(colortype::Type, size::NTuple{2, Int}) = zeros(colortype, size)
+
+generatecanvas(colortype::Type, x::Int, y::Int, b::AbstractBackground) =
+    generatecanvas(colortype, (x,y), b)
+
+function generatecanvas(colortype::Type, size::NTuple{2, Int}, b::AbstractBackground)
+    draw!(zeros(colortype, (size...)), b)
+end
+
+
+"""
+    new_img = draw!(img, solid_background)
+
+    Paints the entire given image a solid color
+"""
+function draw!(img::AbstractArray{T,2}, b::SolidBackground) where {T<:Colorant}
+    b.color == zero(typeof(b.color)) ? (return img) : nothing
+    x, y = size(img)
+    for i = 1:y
+        for j = 1:x
+            draw!(img, Point(j, i), b.color)
+        end
+    end
+    img
+end
+
+"""
+    new_img = draw!(img, striped_background)
+
+    Layers colors onto a given image to set a "background"
+"""
+function draw!(img::AbstractArray{T,2}, b::StripedBackground) where {T<:Colorant}
```

Comment:
This one is interesting and adds novel functionality. However, you seem to implicitly assume that `img` is an all-zeros array, in which case maybe this shouldn't take `img` as an argument? Should we have a function called `stripes`?

Reply:
Also worth asking: how often do you use angles different from 0 and π/2? If your stripes are all rectilinear then there are easier implementations.